lib/page-object/accessors.rb

```diff
@@ -1304,6 +1304,55 @@ def indexed_property (name, identifier_list)
     end
 
     #
+    # adds a method to return a specified class wrapping a generic Element object
+    #
+    # @example
+    #   page_section(:navigation_bar, :id => 'nav-bar')
+    #   # will generate 'navigation_bar' and 'navigation_bar?'
+    #
+    # @param [Symbol] the name used for the generated methods
+    # @param [Hash] identifier how we find an element.  You can use multiple parameters
+    #   by combining of any of the following except xpath.  The valid keys are:
+    #   * :class => Watir and Selenium
+    #   * :css => Selenium only
+    #   * :id => Watir and Selenium
+    #   * :index => Watir and Selenium
+    #   * :name => Watir and Selenium
+    #   * :xpath => Watir and Selenium
+    #
+    def page_section (name, section_class, identifier_list)
+      define_method("#{name}") do
+        platform.page_section_for(section_class, identifier_list.clone)
+      end
+      define_method("#{name}?") do
+        platform.page_section_for(section_class, identifier_list.clone).exists?
+      end
+    end
+
+    #
+    # adds a method to return a collection of specified classes wrapping generic Element objects
+    #
+    # @example
+    #   page_sections(:articles, :class => 'article')
+    #   # will generate 'articles'
+    #
+    # @param [Symbol] the name used for the generated method
+    # @param [Hash] identifier how we find an element.  You can use a multiple parameters
+    #   by combining of any of the following except xpath.  The valid keys are:
+    #   * :class => Watir and Selenium
+    #   * :css => Selenium only
+    #   * :id => Watir and Selenium
+    #   * :index => Watir and Selenium
+    #   * :name => Watir and Selenium
+    #   * :xpath => Watir and Selenium
+    #
+    def page_sections (name, section_class, identifier_list)
+      define_method("#{name}") do
```

Comment:
I decided to use name directly for the method name because although this is a collection, automatic pluralization is hard and prone to error. The user can just supply a name that is their preferred plural form.